cmd/prysmctl/testnet/generate_genesis.go

```diff
@@ -251,43 +217,73 @@ func setGlobalParams() error {
 	return params.SetActive(cfg.Copy())
 }
 
-func generateGenesis(ctx context.Context) (genesisState *ethpb.BeaconState, err error) {
-	genesisTime := generateGenesisStateFlags.GenesisTime
-	numValidators := generateGenesisStateFlags.NumValidators
-	depositJsonFile := generateGenesisStateFlags.DepositJsonFile
-	overrideEth1Data := generateGenesisStateFlags.OverrideEth1Data
-	if depositJsonFile != "" {
-		expanded, err := file.ExpandPath(depositJsonFile)
+func generateGenesis(ctx context.Context) (state.BeaconState, error) {
+	f := &generateGenesisStateFlags
+	if f.GenesisTime == 0 {
+		f.GenesisTime = uint64(time.Now().Unix())
+		log.Info("No genesis time specified, defaulting to now()")
+	}
+
+	v, err := version.FromString(f.ForkName)
+	if err != nil {
+		return nil, err
+	}
+	opts := make([]interop.PremineGenesisOpt, 0)
+	nv := f.NumValidators
+	if f.DepositJsonFile != "" {
+		expanded, err := file.ExpandPath(f.DepositJsonFile)
 		if err != nil {
 			return nil, err
 		}
-		inputJSON, err := os.Open(expanded) // #nosec G304
+		log.Printf("reading deposits from JSON at %s", expanded)
+		b, err := os.ReadFile(expanded)
 		if err != nil {
 			return nil, err
 		}
-		defer func() {
-			if err = inputJSON.Close(); err != nil {
-				log.WithError(err).Printf("Could not close file %s", depositJsonFile)
-			}
-		}()
-		log.Printf("Generating genesis state from input JSON deposit data %s", depositJsonFile)
-		genesisState, err = genesisStateFromJSONValidators(ctx, inputJSON, genesisTime)
+		roots, dds, err := depositEntriesFromJSON(b)
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		if numValidators == 0 {
-			return nil, fmt.Errorf(
-				"expected --num-validators > 0 to have been provided",
-			)
+		opts = append(opts, interop.WithDepositData(dds, roots))
+	}
+
+	if nv == 0 {
+		return nil, fmt.Errorf(
+			"expected --num-validators > 0 or --deposit-json-file to have been provided",
+		)
+	}
+
+	gen := &core.Genesis{}
+	if f.GethGenesisJsonIn != "" {
+		gbytes, err := os.ReadFile(f.GethGenesisJsonIn)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read %s", f.GethGenesisJsonIn)
+		}
+		if err := json.Unmarshal(gbytes, gen); err != nil {
+			return nil, err
 		}
-		// If no JSON input is specified, we create the state deterministically from interop keys.
-		genesisState, _, err = interop.GenerateGenesisState(ctx, genesisTime, numValidators)
+	} else {
+		gen = interop.GethTestnetGenesis(f.GenesisTime, params.BeaconConfig())
+	}
+	if f.GethGenesisJsonOut != "" {
+		gbytes, err := json.Marshal(gen)
 		if err != nil {
 			return nil, err
 		}
+		if err := os.WriteFile(f.GethGenesisJsonOut, gbytes, os.ModePerm); err != nil {
+			return nil, errors.Wrapf(err, "failed to write %s", f.GethGenesisJsonOut)
+		}
 	}
-	if overrideEth1Data {
+
+	gb := gen.ToBlock()
+
+	// TODO: expose the PregenesisCreds option with a cli flag - for now defaulting to no withdrawal credentials at genesis
```

Comment:
Is this worth a TODO+issue? Basically this option allows us to seed the genesis state with withdrawal credentials so that we can test withdrawals without sending bls change messages first.

Reply:
Issue would be good so we don't forget in case we need it in the future and would offer context to people wanting to take it on